tests/libcudacxx_flag_test/libcudacxx_flag_test.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+# Copyright (c) 2025, NVIDIA CORPORATION.
+
+set -euo pipefail
+
+# Get the directory of this script
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+cd "${SCRIPT_DIR}"
+
+# Path to the RMM include directory (absolute path)
+RMM_INCLUDE_DIR="${SCRIPT_DIR}/../../include"
+
+echo "Testing compilation failure when LIBCUDACXX_ENABLE_EXPERIMENTAL_MEMORY_RESOURCE is not defined"
+echo "Using RMM include directory: ${RMM_INCLUDE_DIR}"
+
+# Create a temporary file for compilation errors
+ERROR_FILE=$(mktemp)
+trap 'rm -f "${ERROR_FILE}"' EXIT
+
+# Try to compile the file without defining LIBCUDACXX_ENABLE_EXPERIMENTAL_MEMORY_RESOURCE
+g++ -std=c++17 -I"${RMM_INCLUDE_DIR}" libcudacxx_flag_test.cpp -o libcudacxx_flag_test 2> "${ERROR_FILE}"
+if $?; then
+  echo "Test failed: Compilation succeeded when it should have failed"
+  exit 1
+else
```

Comment:
Very very minor style comment, but I'd remove the nesting here. Invoking `exit` is an early return, there's no need to `else` it.

Like this:

```bash
if $?; then
  echo "Test failed: Compilation succeeded when it should have failed"
  exit 1
fi

if grep -q "RMM requires LIBCUDACXX_ENABLE_EXPERIMENTAL_MEMORY_RESOURCE to be defined" "${ERROR_FILE}"; then
    echo "Test failed: Compilation failed but with an unexpected error message:"
    cat "${ERROR_FILE}"
    exit 1
fi

echo "Test passed: Compilation failed with the expected error message"
exit 0
```

In my opinion, this makes it a little easier to read and to extend with more conditions. But do what you want with that suggestion, it's definitely not blocking and not an opinion I hold strongly.

Reply:
That's a great improvement. I applied this in 50ee60b9b1d8c5d16ec4cf3355d979dc3dad3d3b.